src/helm/benchmark/metrics/basic_metrics.py

```diff
@@ -784,22 +784,21 @@ def compute_logprob_and_length(request_state: RequestState, window_service: Wind
         else:
             raise ValueError(f"Unknown adapter method: {adapter_spec.method}")
 
-        answer_scores = [reference_scores[i] for i in answers]
-
         stats: List[Stat] = []
         stats.extend(self.compute_all_general_metrics(adapter_spec, request_state, metric_service))
 
         max_prob = np.max(scipy.special.softmax(reference_scores))
-        # TODO: fix the accuracy calculation---it is currently incorrect when there are ties.
-        # For example, in binary classification if the model's predicted probabilities are
-        # [0.5, 0.5], then this code will say we got the example correct, regardless of what
-        # the answer is, because the calculation max(reference_scores) == max(answer_scores)
-        # will always return True.
+
+        # It is likely that multiple references achieved the same maximal score. We break ties by picking the
+        # first reference in the list. Exact match is calculated as the fraction of argmax references that are correct.
```

Comment:
The comment about the first reference is out of date.

Reply:
Oh I see, you take the first one for predicted_index...could you please clarify?

Reply:
It was returning `reference_scores.index(max(reference_scores))`. I think `predicted_index` can only contain a single element.

Reply:
I found this comment confusing. I would clarify that the tiebreaking _only_ happens for `predicted_index` and not for `exact_match`. Better yet, break this comment into two by moving the part about tiebreaking later in the file, next to `exact_match`.

Reply:
Improved the comment and hope it is less confusing now.

Reply:
@yifanmai I think it should read better now.